#!/usr/bin/env python
PACKAGE = "synchronizer_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

clock_enum = gen.enum([ gen.const("No_Clock",            int_t, 0, "Not send clock"),
                        gen.const("Computer_Clock",      int_t, 1, "Send computer clock"),
                        gen.const("GPS_Clock",           int_t, 2, "Send gps clock"),
                        gen.const("Manual_Clock",        int_t, 3, "Send manual clock")],
                      "An enum to set different clock")

gen.add("Clock_Type", int_t, 0, "A clock selection for Arduino", 0, 0, 3, edit_method=clock_enum)

exit(gen.generate(PACKAGE, "synchronizer_ros", "Clock"))